api/models/Centres.js

```diff
@@ -90,18 +90,24 @@ const model = {
       return response;
     }
   },
+
   getGenderAndCentreByCIDLocation: function (location) {
-    return this.findOne(
-      {
-        or: [
-          {male_cid_name: location},
-          {female_cid_name: location}
-        ]
-      })
-      .then(centre => centre ? {
-        centre: centre.id,
-        gender: centre.male_cid_name === location ? "male" : "female"
-      } : centre);
+    return this.find().then(centres =>
+      _.compact(_.map(centres, centre => {
```

Comment:
does waterline/sails cache this? if not is there a better way? of keeping a cache of the `find()` for the current execution to avoid getting all the centres a few hundred times every few minutes?
We'd talked briefly about producing some sort of cacheable map of:
`{
'locationa': {centre:1, gender:'male'},
'locationb': {centre:1, gender:'male'}
//etc
};
`